controllers/argocd/status.go

```diff
@@ -212,3 +217,44 @@ func (r *ReconcileArgoCD) reconcileStatusServer(cr *argoprojv1a1.ArgoCD) error {
 	}
 	return nil
 }
+
+// reconcileStatusURL will ensure that the URL status is updated for the given ArgoCD.
+func (r *ReconcileArgoCD) reconcileStatusURL(cr *argoprojv1a1.ArgoCD) error {
+
+	if cr.Spec.Server.Route.Enabled {
+		if !IsRouteAPIAvailable() {
+			log.Info("Routes not available in non-OpenShift environments, please use Ingresses instead")
+			return nil
+		}
+		route := newRouteWithSuffix("server", cr)
+		if argoutil.IsObjectFound(r.Client, cr.Namespace, route.Name, route) {
+			if route.Status.Ingress == nil {
+				status := "Pending"
+				cr.Status.URL = status
+				cr.Status.Phase = status
+				return r.Client.Status().Update(context.TODO(), cr)
+			} else {
+				cr.Status.URL = route.Spec.Host
```

Comment:
We should get the host from `route.Status.Ingress` which is a list of RouteIngresses. I think we should show all the ingresses.